src/GafferImage/ImageWriter.cpp

```diff
@@ -869,6 +935,12 @@ IECore::MurmurHash ImageWriter::hash( const Context *context ) const
 void ImageWriter::execute() const
 {
 	// Create an OIIO::ImageOutput
+	Context * c = const_cast<Context *>( Context::current() );
+	c->set( "gaffer:colorspace", defaultColorSpace()( currentFileFormat() ) );
```

Comment:
Yeah, he's talking about a potential future node.  It would be quite useful - there are places in my current work where I have to set loop indices to 0 instead of deleting them, which means that I'm unnecessarily hashing things twice ( I'd be hashing them a lot more than that if I didn't reset the index ).

So, yeah, DeleteContextVariables sounds like a great idea.